File: Bit_Manipulation/338_counting_bits/338.c


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* countBits(int n, int* returnSize){
    
    *returnSize = n + 1;
    int* result = (int*)malloc((n + 1) * sizeof(int));
    result[0] = 0;
    for (int i = 1; i < (n + 1); i++) {
        result[i] = 0;
        result[i] = result[i & (i - 1)] + 1;
    }
    return result;
}

File: Bit_Manipulation/338_counting_bits/338.rs


impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        let mut result: Vec<i32> = vec![0;n as usize + 1];
        for i in 1..(n as usize + 1) {
            result[i] = result[i & (i - 1)] + 1;
        }
        return result;
    }
}